stream/stream_dvbv5.c

```diff
@@ -0,0 +1,577 @@
+        dvb_dev_close(sid_fd);
+        if (pmtpid <= 0) {
+            MP_ERR(stream, "couldn't find pmt-pid for sid %04x\n",
+                sid);
+
+            dvb_dev_free(dvb);
+            return STREAM_ERROR;
+        }
+
+        pat_fd = dvb_dev_open(dvb, demux_dev, O_RDWR);
+        if (!pat_fd) {
+            MP_ERR(stream, "opening pat demux failed");
+            dvb_dev_free(dvb);
+            return STREAM_ERROR;
+        }
+        if (dvb_dev_dmx_set_pesfilter(pat_fd, 0, DMX_PES_OTHER,
+                DMX_OUT_TS_TAP,
+                64 * 1024) < 0) {
+            dvb_dev_free(dvb);
+            return STREAM_ERROR;
+        }
+
+        pmt_fd = dvb_dev_open(dvb, demux_dev, O_RDWR);
+        if (!pmt_fd) {
+            MP_ERR(stream, "opening pmt demux failed");
+            dvb_dev_free(dvb);
+            return STREAM_ERROR;
+        }
+        if (dvb_dev_dmx_set_pesfilter(pmt_fd, pmtpid, DMX_PES_OTHER,
+                DMX_OUT_TS_TAP,
+                64 * 1024) < 0) {
+            dvb_dev_free(dvb);
+            return STREAM_ERROR;
+        }
+
+        /*
+         * SDT may also be needed in order to play some streams
+         */
+        sdt_fd = dvb_dev_open(dvb, demux_dev, O_RDWR);
+        if (!sdt_fd) {
+            MP_ERR(stream, "opening sdt demux failed");
+            dvb_dev_free(dvb);
+            return STREAM_ERROR;
+        }
+        if (dvb_dev_dmx_set_pesfilter(sdt_fd, 0x0011, DMX_PES_OTHER,
+                DMX_OUT_TS_TAP,
+                64 * 1024) < 0) {
+            dvb_dev_free(dvb);
+            return STREAM_ERROR;
+        }
+    }
+
+      if (vpid >= 0) {
+        video_fd = dvb_dev_open(dvb, demux_dev, O_RDWR);
+        if (!video_fd) {
+            MP_ERR(stream, "failed opening '%s'", demux_dev);
+            dvb_dev_free(dvb);
+            return STREAM_ERROR;
+        }
+
+        dvb_dev_set_bufsize(video_fd, DVB_BUF_SIZE);
+
+        if (dvb_dev_dmx_set_pesfilter(video_fd, vpid, DMX_PES_VIDEO,
+            DMX_OUT_TS_TAP,
+            64 * 1024) < 0) {
+            dvb_dev_free(dvb);
+            return STREAM_ERROR;
+        }
+    }
+
+    if (apid > 0) {
+        audio_fd = dvb_dev_open(dvb, demux_dev, O_RDWR);
+        if (!audio_fd) {
+            MP_ERR(stream, "failed opening '%s'", demux_dev);
+            dvb_dev_free(dvb);
+            return STREAM_ERROR;
+        }
+        if (dvb_dev_dmx_set_pesfilter(audio_fd, apid, DMX_PES_AUDIO,
+                DMX_OUT_TS_TAP,
+                64 * 1024) < 0) {
+            dvb_dev_free(dvb);
+            return STREAM_ERROR;
+        }
+    }
+
+    pthread_create(&timeout, NULL, do_timeout, priv);
+
+    if (!check_frontend(stream, parms)) {
+
+        MP_ERR(stream, "frontend doesn't lock");
+        dvb_dev_free(dvb);
+        return STREAM_ERROR;
+    }
+
+    pthread_cancel(timeout);
+
+    dvr_fd = dvb_dev_open(dvb, dvr_dev, O_RDONLY);
+    if (!dvr_fd) {
+        MP_ERR(stream, "failed opening '%s'", dvr_dev);
```

Comment:
many MP_ERR usages in this file are missing the trailing newline

Reply:
Nice find! That's why I didn't see the output of some MP_ERR messages while testing!

Fixed in [force-push](https://github.com/mpv--player/mpv/compare/b788e75a772b0b17b23b959ccfbb6a97a6f3a457..88a75c225cd0e6a2f5aa98946862d59303f54557)